lib/puppet/parser/environment_compiler.rb

```diff
@@ -73,6 +98,35 @@ def add_resource(scope, resource)
     end
   end
 
+
+  def evaluate_ast_node()
+    # Do nothing, the environment catalog is not built for a particular node.
+  end
+
+  # Evaluates the site - the top container for the environment catalog
+  # The site contain behaves analogous to a node - for the environment catalog node expressions are ignored
+  # as the result is cross node. The site expression serves as a container for everything that is across
```

Comment:
Would be helpful with a comma here. I assume you mean that "node expressions" are ignored in the "environment catalog"? Other interpretations can be "catalog node expressions" in the "environment" or "expressions" in the "environment catalog node" :-)